main.tf

```
@@ -25,6 +25,22 @@ locals {
     id             = coalesce(var.launch_template_id, aws_launch_template.default[0].id)
     latest_version = coalesce(var.launch_template_version, aws_launch_template.default[0].latest_version)
   }
+
+  eks_node_group_resources = [for resource in aws_eks_node_group.default.*.resources :
+    [for i in resource :
+      merge(i, {
+        remote_access_security_group_id = join("", aws_security_group.remote_access.*.id)
+      })
+    ]
+  ]
+
+  security_group_ids = concat(
+    list(data.aws_eks_cluster.eks_cluster.vpc_config[0].cluster_security_group_id),
+    var.ec2_ssh_key != null ? aws_security_group.remote_access.*.id : []
+  )
+
```

I am still figuring out the best way to allow variables to be empty but easy to test. If you have a better idea, I would love to hear it; for now what I have is:
```suggestion
    local.ec2_ssh_key != null ? aws_security_group.remote_access.*.id : []
  )

  ec2_ssh_key = var.ec2_ssh_key == "" ? : null ? var.ec2_ssh_key
```